compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -275,6 +275,16 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
     case _ =>
   }
 
+  /** Make `tree` part of inlined expansion. This means its owner has to be changed
+   *  from its `originalOwner`, and, if it comes from outside the inlined method
+   *  itself, it has to be marked as an inlined argument.
+   */
+  def integrate(tree: Tree, originalOwner: Symbol)(implicit ctx: Context) = {
+    val result = tree.changeOwner(originalOwner, ctx.owner)
+    if (!originalOwner.isContainedIn(inlinedMethod)) Inlined(EmptyTree, Nil, result)
```

Comment:
Repurposing the `Inlined` tree here seems weird, I would use an annotation instead.

Reply:
The problem is that everybody has to know to update their contexts when traversing into one of these nodes. It's too easily forgotten if this is an annotation. If this is an Inline code then

 - either some code will not update contexts either for Inlined code or for inlined arguments. That's OK as long as positions don't have to be reported accurately.
 - or some code will update contexts for both.

We avoid the annoying case by design where code will only update context for inlined code but forget to undo the update for inlined arguments.